### PR TITLE
Update ubuntu2004 image with deb and gh packages

### LINKED ISSUE #28: OpenSearch / OpenSearch Dashboards - DEB distribution (X64)
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
Provide signed (X64)Debian artifacts for OpenSearch / OpenSearch Dashboards 

**Describe alternatives you've considered**
Generating the artifacts using the source code 

**Additional context**
Currently there are no DEB artifacts for OpenSearch / OpenSearch Dashboards 

- [ ] Refresh Button

------

* Design: #2460 
* PRs:
  * #2526 
  * https://github.com/opensearch-project/opensearch-build/pull/2950
  * https://github.com/opensearch-project/opensearch-build/pull/3060
  * #3090 
  * https://github.com/opensearch-project/opensearch-build-libraries/pull/109
  * https://github.com/opensearch-project/opensearch-build/pull/3095

### PR DESCRIPTION
**Pull request by peterzhuamazon**
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

### Description
Update ubuntu2004 image with deb and gh packages

### Issues Resolved
#28

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docker/ci/dockerfiles/current/docker-builder.ubuntu2004.x64.dockerfile [hunk 1]
```diff
@@ -54,3 +56,9 @@ RUN curl -L https://github.com/google/go-containerregistry/releases/latest/downl
     chmod +x gcrane && \
     mv gcrane /usr/local/bin/ && \
     rm -rf go-containerregistry.tar.gz 
+
+# Install gh-cli
```
Comment (gaiksaya) [inline docker/ci/dockerfiles/current/docker-builder.ubuntu2004.x64.dockerfile:60]: What do we need gh-cli for?

Comment (peterzhuamazon) [reply docker/ci/dockerfiles/current/docker-builder.ubuntu2004.x64.dockerfile:60]: Basically every image now have gh-cli for creating issues on GitHub. @rishabh6788 

I am just adding this up which does not take too much space.

If you think it is not needed we can always remove.

**Review by gaiksaya — APPROVED** [review]

File: docker/ci/dockerfiles/current/docker-builder.ubuntu2004.x64.dockerfile (hunk 1 above)
Comment (gaiksaya) [reply docker/ci/dockerfiles/current/docker-builder.ubuntu2004.x64.dockerfile:60]: Looks like the post actions runs on agent node directly https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/opensearch/distribution-build.jenkinsfile#L879
@rishabh6788 can add more. Remove if not required.

Comment (rishabh6788) [reply docker/ci/dockerfiles/current/docker-builder.ubuntu2004.x64.dockerfile:60]: Yes, it is run on the node directly, not required inside the docker image. @peterzhuamazon @gaiksaya

Comment (peterzhuamazon) [reply docker/ci/dockerfiles/current/docker-builder.ubuntu2004.x64.dockerfile:60]: This is more of a per-requisite in case we need to use this for gh cli.

Comment (peterzhuamazon) [reply docker/ci/dockerfiles/current/docker-builder.ubuntu2004.x64.dockerfile]: Removed now.

**Review by gaiksaya — APPROVED** [review]

**Review by rishabh6788 — APPROVED** [review]

**Comment by peterzhuamazon** [comment]
https://build.ci.opensearch.org/job/docker-build/2106
https://build.ci.opensearch.org/job/docker-copy/219

**Pull request merged**